drltrace/scripts/gen_drltrace_config.py

```diff
@@ -0,0 +1,537 @@
+                element = element.replace("*", "")
+            type = get_type(element) # if type is unknown we have None string
+            if type == "None":
+                try:
+                    unknown_types[element] += 1
+                except:
+                    unknown_types[element] = 1
+            else:
+                res_type = type
+        if res_type != "None":
+            if param_type == OUT_TYPE:
+                res_type = "+" + res_type # equals __out
+            elif param_type ==  INOUT_TYPE:
+                res_type = "++" + res_type # equals __inout
+            if pointer == True:
+                res_type += "*"
+        res_types.append(str(res_type))
+    return_string = "|".join(res_types)
+    return_string = return_string.replace("++", "__inout ")
+    return_string = return_string.replace("+", "__out ")
+    return return_string
+
+def parse_line(line, name):
+    '''
+    The function starts parsing of the line specified.
+    @in line - a line to parse.
+    @in name - a name of WinAPI function.
+    @out - a string with a function return type, name and arguments separated by |.
+    '''
+
+    prefix = line[:line.find(name)]
+    suffix = line[line.find(name)+len(name) + 1:]
+    new_prefix = parse_return_type(prefix)
+    args = parse_args(suffix)
+    return "%s|%s|%s" % (new_prefix, name, args)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description = "The script is used for searching of"
+" WinAPI function prototypes in the output of headers_parser.py and converting them into"
+" the format supported by drltrace.")
+    parser.add_argument('-results_path', help = "The path where headers_parser.py saved results.",
+    default = "results\\")
+    parser.add_argument('-parse_specific_file', help = "The option is used to specify for"
+" the script only a single file")
+
+    args = parser.parse_args()
+
+    export_files = list()
+    if args.parse_specific_file == None:
+        print "Parsing all files from %s" % args.results_path
+        export_files = [f for f in listdir(args.results_path) if isfile(join(args.results_path, f))]
+    else:
+        print "Parsing %s from %s" % (args.parse_specific_file, args.results_path)
+        if not args.parse_specific_file.endswith(".headers_out"):
+            print "File should end with *.headers_out"
+            sys.exit(-1)
+        export_files.append(args.parse_specific_file)
+
+    for file in export_files:
+        if not file.endswith(".headers_out"): # parse only output of headers_parser.py
+            continue
+        content = open(args.results_path + file, 'r').readlines()
+        file_write = open(file + ".config_raw", 'w') # the output file
+        file_write.write("The result is not a final config. Manual analysis is required\n")
+
+        for line in content:
+            line = line[:-1]
+            # the headers_parser.py saves possible prototypes using the following pattern:
```

Comment:
Un-capitalized and un-punctuated comments should be limited to short one-line phrases: anything long like this should be capitalized and punctuated (and for short we prefer that as well for all new code).